src/common/pool_hdr.h

```diff
@@ -265,12 +265,13 @@ static const features_t features_zero =
  * POOL_FEAT_CKSUM_2K incompat feature is set.
  */
 #define POOL_HDR_CSUM_END_OFF(hdrp) \
-	((hdrp)->features.incompat & POOL_FEAT_CKSUM_2K) \
+	(le32toh((hdrp)->features.incompat) & POOL_FEAT_CKSUM_2K) \
```

Comment:
Ok, let's see if this is called on a header in little-endian order:
util_header_create - no, because you moved util_convert2le_hdr after POOL_HDR_CSUM_END_OFF
util_header_check - no, util_convert2h_hdr_nocheck now always converts to host order
util_header_check_remote - no, util_convert2h_hdr_nocheck is called at the beginning
util_replica_set_attr - yes
pool_hdr_valid - not sure
pool_hdr_checksum_fix - yes
set_hdr - no, util_convert2le_hdr is called immediately after POOL_HDR_CSUM_END_OFF
check_checksums_and_signatures - not sure
update_parts_linkage - not sure
update_replicas_linkage - not sure
update_poolset_uuids - not sure
update_replica_header - probably no, few lines above we access hdr->features.incompat as it is in host order
update_uuids - not sure

It seems pmdk needs deep endianness audit and some cleanups to untangle this :(.

Reply:
I don't think this is doable without some extensive testing — human audit is needed, but there's no way we'd get it working by just reading code.  Thus: do you have a test setup on a big-endian machine that's capable of running a good subset of functionality of higher-layer libraries?  At least your other pull request is little-endian only.

Also, I wonder what's the reason for supporting big-endian.  I see no kernel support for anything pmem on s390, and I got the impression IBM is not just deprecating but even actively pushing other projects to drop support for big-endian PPC.  I'm not sure if the impression is true — it may be more about distribution baselines for ppc64be being POWER4 (ppc64el is POWER8, a last-minute bump from POWER7 before final submission on the behest of IBM folks).

So unless some extensive testing can be done, I have doubts about any chances to get it right.

Reply:
> It seems pmdk needs deep endianness audit and some cleanups to untangle this :(.

>So unless some extensive testing can be done, I have doubts about any chances to get it right.

I agree that a porting this to BE should be done with an wider audit and a huge among of
tests. I actually caught this bug because I accidentally build PMDK on a BE machine, I was meant to use a LE instead.

It's not my intention to add BE support in pmdk. So I think it's better to close this PR and postpone this effort for some time, until BE is really needed, if at all.
Do you agree?

Reply:
I think this is the correct approach.

Reply:
I mean, postponing until there is clear requirement is the correct approach.